File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int ans = Integer.MAX_VALUE, psum = 0;
        int i = 0;

        for (int j = 0; j < nums.length; j++) {
            psum += nums[j];
// the minimal length of a subarray whose sum is greater than or equal to target
            while (psum >= target) {
                psum -= nums[i];
                ans = Math.min(ans, j - i + 1);
                i++;
            }
        }
        return ans == Integer.MAX_VALUE ? 0 : ans;
    }
}

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, t: int, nums: List[int]) -> int:
        n = len(nums)
        i = j = sum_t = 0
        ans = 1000000000

        while j < n:
            sum_t += nums[j]

            while sum_t >= t:
                sum_t -= nums[i]
                ans = min(j - i + 1, ans)
                i += 1
            j += 1

        return 0 if ans == 1000000000 else ans
